solidity-v1/dashboard/src/components/modal/threshold/WithdrawGrantedTokens.jsx

```diff
@@ -0,0 +1,201 @@
+  const DEFAULT_GRANTS_TO_DISPLAY = 5
+  const [numberOfGrantsDisplayed, setNumberOfGrantsDisplayed] = useState(
+    DEFAULT_GRANTS_TO_DISPLAY
+  )
+  const [selectedGrant, setSelectedGrant] = useState(
+    grants.length === 1 ? grants[0].id : null
+  )
+  const releaseTokens = useReleaseTokens()
+
+  const totalReadyToRelease = useMemo(() => {
+    return grants
+      .map((grant) => grant.readyToRelease)
+      .reduce((previous, current) => add(previous, current), 0)
+      .toString()
+  }, [grants])
+
+  const onWithdrawClick = (awaitingPromise) => {
+    releaseTokens(selectedGrant, awaitingPromise)
+  }
+
+  return (
+    <>
+      <ModalHeader>Withdraw</ModalHeader>
+      <ModalBody>
+        <h3 className={"mb-1"}>Withdraw granted tokens</h3>
+        <p className={"mb-2 text-grey-70"}>
+          The following tokens are available to be released and withdrawn from
+          your token grant.
+        </p>
+        <div>
+          <OnlyIf condition={grants.length > 1}>
+            <div className="flex row center space-between">
+              <h4 className="text-grey-70">Withdraw:</h4>
+              <span className="text-grey-70 small">
+                Total:{" "}
+                <TokenAmount
+                  amount={totalReadyToRelease}
+                  token={KEEP}
+                  wrapperClassName="withdraw-granted-tokens__total-ready-to-release-amount"
+                  amountClassName="text-grey-60"
+                  symbolClassName="text-grey-60"
+                />
+              </span>
+            </div>
+          </OnlyIf>
+          <form>
+            <Accordion
+              allowZeroExpanded={grants.length > 1}
+              className={
+                "withdraw-granted-tokens__grants-accordion withdraw-granted-tokens__grants-accordion-container"
+              }
+              preExpanded={grants.length === 0 ? [`grant-${grants[0].id}`] : []}
+            >
+              {grants
+                .slice(0, numberOfGrantsDisplayed)
+                .map((grant) =>
+                  renderGrant(
+                    grant,
+                    grants.length,
+                    selectedGrant,
+                    setSelectedGrant
+                  )
+                )}
+            </Accordion>
+          </form>
+          <OnlyIf condition={grants.length > numberOfGrantsDisplayed}>
+            <Button
+              className="withdraw-granted-tokens__view-more-button"
+              onClick={() =>
+                setNumberOfGrantsDisplayed(
+                  numberOfGrantsDisplayed + DEFAULT_GRANTS_TO_DISPLAY
+                )
+              }
+            >
+              <Icons.Add />
+              &nbsp;View More
+            </Button>
+          </OnlyIf>
+        </div>
+      </ModalBody>
+      <ModalFooter>
+        <SubmitButton
+          className="btn btn-primary btn-lg mr-2"
+          type="submit"
+          onSubmitAction={(awaitingPromise) => {
```

Comment:
Let's just `onSubmitAction={onWithdrawClick}`.

Reply:
Fixed in 0f11cc1db404ce9fade9e48fabeb4bc1cc144da3